src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -316,6 +316,15 @@
          of unsafe code - such as fixed field indexing. -->
     <Field Name="IsManaged" Type="bool"/>  
   </Node>
+  <!-- Represents an AddressOf operator that has not yet been assigned to a
+       target-type. It has no natural type, and should not survive initial binding. -->
+  <Node Name="BoundUnconvertedAddressOfOperator" Base="BoundExpression">
+    <Field Name="Operand" Type="BoundMethodGroup"/>
+  </Node>
+  <!-- Represents a resolved AddressOf function pointer. Not used in initial binding. -->
+  <Node Name="BoundFunctionPointerLoad" Base="BoundExpression">
+    <Field Name="TargetMethod" Type="MethodSymbol"/>
+  </Node>
```

Comment:
></Node> [](start = 2, length = 7)

I think for the benefit of SemanticModel we should preserve the BoundMethodGroup from the node above. Please add targeted SemanticModel tests for the new nodes. TypeInfo, SymbolInfo, etc.

Reply:
> I think for the benefit of SemanticModel we should preserve the BoundMethodGroup from the node above.

The `SemanticModel` will never see a `BoundFunctionPointerLoad` node. It is only introduced during lowering stages.

Reply:
IOperation as well.

---
In reply to: [394000465](https://github.com/dotnet/roslyn/pull/42488#discussion_r394000465) [](ancestors = 394000465)

Reply:
> Please add targeted SemanticModel tests for the new nodes. TypeInfo, SymbolInfo, etc.

I will add a prototype comment to add targeted tests for BoundUnconvertedAddressOfOperator. There's also `BoundFunctionPointerInvocation` that needs to be tested as well, as well as symbol display tests, exposing calling conventions publicly, and a few more items. My plan is to address these all together at a later time.